doc/develop/west/build-flash-debug.rst

```diff
@@ -576,6 +679,38 @@ For example, to print usage information about the ``jlink`` runner::
 
   west debug -H -r jlink
 
+.. _west-multi-domain-debugging:
+
+Multi-domain debugging
```

Comment:
Do we really want to be recommending use of `west debug` here? I think that in a multi-image context, `west attach` is probably the better command to be using. Since `attach` doesn't reflash anything, it can't mess anything up, while `debug` can reflash things, which might result in odd states on flash if domains need to be flashed in a specific sequence.

Reply:
but if you're working on a sample that enables MCUBoot, then you probably want to have the application re-flashed before debugging if you made code changes.

Which is also the reason the main application (which is the one you pointed to at build time) is the application that default will be used for debugging.

I don't claim to know the right answer here, cause it's hard to predict exactly how sysbuild is used for those purposes, but this behavior is somewhat similar to the existing behavior in the nRF Connect SDK multi image.
And I haven't heard complaints on that part. So instead of recommending two different approaches, one approach when not using sysbuild, another when using sysbuild, then I felt it safest to just keep the existing way.

Reply:
> And I haven't heard complaints on that part.

I have had to help at least one user who got confused about this. But it's OK, let's leave it as is for now and we can adjust depending on the complaints we get from users.

Reply:
I've been thinking a bit.

I still think it makes sense to describe how `west debug` functions, cause if that is what users are used to, it's probably a workflow they will continue to use.

But I think we can address the surprises of `west debug` in a multi domain setup, and instead highlight the possibility of `west attach` for this scenario as a good (better !?) alternative in this scenario.

But I will focus on the other comments first, and the `attach` addition can be introduced later if need be.